pkg/diff/diff.go

```diff
@@ -120,6 +130,126 @@ func Diff(config, live *unstructured.Unstructured, opts ...Option) (*DiffResult,
 	return TwoWayDiff(config, live)
 }
 
+func ServerSideDiff(config, live *unstructured.Unstructured, opts ...Option) (*DiffResult, error) {
+	if live != nil && config != nil {
+		return serverSideDiff(config, live, opts...)
+	}
+	return handleResourceCreateOrDeleteDiff(config, live)
```

Comment:
I think it is fine to keep the existing behaviour and display a shallow diff in case of "create". A few reasons that comes to my mind:
- Saves 1 additional call to KubeAPI
- Much lighter/faster diff
- This is the existing behaviour users are already used to

Reply:
Makes sense. I'd maybe just add a comment here explaining that decision and the tradeoff.

Reply:
done